contract/models/abstract_contract_line.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2004-2010 OpenERP SA
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# Copyright 2015 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# Copyright 2016-2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+
+
+class ContractAbstractContractLine(models.AbstractModel):
+    _inherit = "contract.recurrency.basic.mixin"
+    _name = "contract.abstract.contract.line"
+    _description = "Abstract Recurring Contract Line"
+
+    product_id = fields.Many2one("product.product", string="Product")
+
+    name = fields.Text(string="Description", required=True)
+    quantity = fields.Float(default=1.0, required=True)
+    product_uom_category_id = fields.Many2one(  # Used for domain of field uom_id
+        comodel_name="uom.category",
+        related="product_id.uom_id.category_id",
+    )
+    uom_id = fields.Many2one(
+        comodel_name="uom.uom",
+        string="Unit of Measure",
+        domain="[('category_id', '=', product_uom_category_id)]",
+    )
+    automatic_price = fields.Boolean(
+        string="Auto-price?",
+        help="If this is marked, the price will be obtained automatically "
+        "applying the pricelist to the product. If not, you will be "
+        "able to introduce a manual price",
+    )
+    # Just to have a currency_id here - will get overwriten in contract.line
+    # model with the related currency from the contract
+    currency_id = fields.Many2one("res.currency")
+    specific_price = fields.Monetary()
+    price_unit = fields.Monetary(
+        string="Unit Price",
+        compute="_compute_price_unit",
+        inverse="_inverse_price_unit",
+    )
```

Comment:
@pedrobaeza what do you think about changing this to float regarding this conversation: https://github.com/OCA/account-invoicing/pull/1674#pullrequestreview-1975650402

Reply:
OK, agreed.